spikeinterface/sortingcomponents/waveforms/pca.py

```diff
@@ -0,0 +1,175 @@
+    2) The class can be trained with the fit method after instantiating the class. The model will be stored in 
+    `model_path`.
+
+
+    Parameters
+    ----------
+    recording : BaseRecording
+        The recording object.
+    parents: list
+        The parent nodes of this node. This should contain a mechanism to extract waveforms.
+    ms_before : float, optional
+        The number of milliseconds to include before the peak of the spike, by default 1.
+    ms_after : float, optional
+        The number of milliseconds to include after the peak of the spike, by default 1.
+    peak_sign : str, optional
+        The sign of the peak, either 'neg' or 'pos', by default 'neg'.
+    all_channels : bool, optional
+        Whether to extract spikes from all channels or only the channels with spikes, by default True.
+    model_path : str, optional
+        The path to the pca model, by default None.
+    local_radius_um : float, optional
+        The radius (in micrometers) to use for definint sparsity, by default None.
+        
+    """    
+
+
+    def __init__(self, recording: BaseRecording, parents: list = list[PipelineNode], ms_before: float = 1., ms_after: float = 1.,
+        peak_sign: str = 'neg', all_channels: bool = True, model_path: str = None, local_radius_um: float = None,
+        return_ouput=True,
+        ):
+        
+        PCBaseNode.__init__(self, recording=recording, parents=parents, ms_before=ms_before, ms_after=ms_after,
+                            peak_sign=peak_sign, all_channels=all_channels, model_path=model_path, 
+                            local_radius_um=local_radius_um, return_ouput=return_ouput)
+            
+    def compute(self, traces: np.ndarray, peaks: np.ndarray, waveforms: np.ndarray) -> np.ndarray:
+        """
+        Projects the waveforms using the PCA model trained in the fit method or loaded from the model_path.
+
+        Parameters
+        ----------
+        traces : np.ndarray
+            The traces of the recording.
+        peaks : np.ndarray
+            The peaks resulting from a peak_detection step.
+        waveforms : np.ndarray
+            Waveforms extracted from the recording using a WavefomExtractor node.
+
+        Returns
+        -------
+        np.ndarray
+            The projected waveforms.
+
+        """
+
+        if self.pca_model == None:
+            exception_string = (
+                f"Pca model not found, "
+                f"Train the pca model before running compute_buffer using {self.__name__}.fit(kwargs)"
+            )
+            raise AttributeError(exception_string)
+        
+        num_waveforms, num_samples , num_channels = waveforms.shape
+        
+        channeless_waveform = to_temporal_representation(waveforms)
+        projected_chaneless_waveforms = self.pca_model.transform(channeless_waveform)
+        projected_waveforms = from_temporal_representation(projected_chaneless_waveforms, num_channels)
+
+        return projected_waveforms
```

Comment:
Should we add a `tPCADenoiser` class to return clean waveforms after backprojection?

Reply:
we could also add it later :)